app/src/main/java/com/mapbox/navigation/examples/alternative/ShowAlternativeRoutesActivity.kt

```diff
@@ -256,7 +284,31 @@ class ShowAlternativeRoutesActivity : AppCompatActivity() {
                         .build()
                 )
                 .build()
-        )
+        ).attach(this)
+
+        lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                MapboxNavigationApp.registerObserver(navigationCreatedObserver)
```

Comment:
Why don't you call this in Activity's `onCreate`?

Reply:
This is not done yet, but a big part of the reason to use `MapboxNavigationApp` is to make it so we can change `NavigationOptions`

For example, if someone calls [requestRoadGraphDataUpdate](https://github.com/mapbox/mapbox-navigation-android/blob/9073234c2978c019980ca606b16fa0d3ff37f85b/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L841-L860) they need to refresh the `MapboxNavigation` instance.

If something hooks up to mapboxNavigation in an `onCreate` function, it will be forgotten when the MapboxNavigation instance changes.

Reply:
I'd like to add a simple way to change `MapboxNavigation` from the command line so we can test it easy.

Reply:
Like this https://github.com/mapbox/mapbox-navigation-android/pull/6234

Reply:
I mean there is no difference at all between using `lifecycle.addObserver` or adding logic directly to lifecycle methods of `Activity`.

Reply:
oh right, getting too used to `lifecycle.addObserver`